src/promptflow-tools/promptflow/tools/open_source_llm.py

```diff
@@ -51,6 +51,100 @@ def wrapper(*args, **kwargs):
     return deco_retry
 
 
+class Endpoint:
+    def __init__(self,
+                 endpoint_name: str,
+                 endpoint_url: str,
+                 endpoint_key: str):
+        self.deployments: List[Deployment] = []
+        self.default_deployment: Deployment = None
+        self.endpoint_url = endpoint_url
+        self.endpoint_key = endpoint_key
+        self.endpoint_name = endpoint_name
+
+
+class Deployment:
+    def __init__(self,
+                 deployment_name: str,
+                 model_family: str):
+        self.model_family = model_family
+        self.deployment_name = deployment_name
+
+
+class EndpointsContainer:
+    '''We're probably going to need to cache this somehow'''
+    def __init__(self) -> None:
+        self.__endpoints_and_deployments = None
+
+    def get_ml_client(self) -> MLClient:
+        from azure.identity import DefaultAzureCredential
+        credential = DefaultAzureCredential(exclude_interactive_browser_credential=False)
+        try:
+            return MLClient(
```

Comment:
early fyi, may not need to set os env, in local user just need to az login and configure azure ws info.

System will append ws triple to list function input params if the function signature has 'subscription_id','resource_group_name','workspace_name' or `**kwargs`. More references:
https://github.com/microsoft/promptflow/blob/a589f8281aa5fd22379c5bff74c968af0bf4b66a/src/promptflow/promptflow/_core/tools_manager.py#L181